wasm/src/programs/decrypt.rs

```diff
@@ -0,0 +1,255 @@
+            let entry = match entry {
+                // Constant entries do not need to be decrypted.
+                Entry::Constant(plaintext) => Entry::Constant(plaintext.clone()),
+                // Public entries do not need to be decrypted.
+                Entry::Public(plaintext) => Entry::Public(plaintext.clone()),
+                // Private entries are decrypted with the given randomizers.
+                Entry::Private(private) => Entry::Private(Plaintext::from_fields(
+                    &private
+                        .iter()
+                        .zip_eq(randomizers)
+                        .map(|(ciphertext, randomizer)| *ciphertext - randomizer)
+                        .collect::<Vec<_>>(),
+                )?),
+            };
+            // Insert the decrypted entry.
+            if decrypted_data.insert(*id, entry).is_some() {
+                bail!("Duplicate identifier in record: {}", id);
+            }
+            // Increment the index.
+            index += num_randomizers;
+        }
+
+        // Return the decrypted record.
+        RecordPlaintext(RecordPlaintextNative{owner, decrypted_data, **record.nonce()})
+    }
+
+    #[wasm_bindgen(js_name = "generateTvk")]
+    pub fn generate_tvk(
+        view_key: &ViewKey,
+        tpk: &Group,
+    ) -> Field {
+        tpk.scalar_multiply(&view_key.to_scalar()).to_x_coordinate()
+    }
+
+    #[wasm_bindgen(js_name = "generateRecordVk")]
+    pub fn generate_record_vk(
+        view_key: &str,
+        record: &str,
+    ) -> Group {
+        let view_key = ViewKey::from_str(view_key)
+            .map_err(|_| "Failed to parse view key".to_string())
+            .unwrap();
+        let record_ciphertext = RecordCiphertext::from_str(record)
+            .map_err(|_| "Failed to parse record ciphertext".to_string())
+            .unwrap();
+        let record_nonce = record_ciphertext.0.nonce();
+        record_nonce * **view_key
+    }
+
+    #[wasm_bindgen(js_name = "decryptRecordWithRVk")]
+    pub fn decrypt_record_symmetric_unchecked(
+        record_vk: &str,
+        record: &str,
+    ) -> Result<RecordPlaintext, String> {
+        let record_vk = Field::from_str(record_vk)
+            .map_err(|_| "Failed to parse record view key".to_string())?;
+        let record_ciphertext = RecordCiphertext::from_str(record)
+            .map_err(|_| "Failed to parse record ciphertext".to_string())?;
+        let num_randomizers = DecryptToolBox::num_randomizers(record_ciphertext)
+            .map_err(|_| "Failed to get the number of randomizers from the record ciphertext".to_string())?;
+        let randomizers = CurrentNetwork::hash_many_psd8(&[CurrentNetwork::encryption_domain(), *record_vk], num_randomizers);
+
+        let record_plaintext = record_ciphertext.decrypt_with_randomizers(&randomizers);
+    
+        Ok(record_plaintext)
+    }
+
+    #[wasm_bindgen(js_name = "decryptTransitionWithVk")]
+    pub fn decrypt_transition_with_vk(
```

Comment:
Instead of `vk` ensure everything in the method name/signature is `view_key` (or `ViewKey` in terms of the `js_name`)

```suggestion
    pub fn decrypt_transition_with_view_key(
```

Reply:
done [ccd1c57](https://github.com/ProvableHQ/sdk/pull/1049/commits/ccd1c57cd2239a9f8c79f01e8b920f645eccb595)